src/sagemaker_training/process.py

```diff
@@ -177,20 +183,22 @@ def check_error(cmd, error_class, processes_per_host, cwd=None, capture_error=Tr
         subprocess.Popen: The process for the given command.
 
     Raises:
-        error_class: If there is an exception raised when creating the process.
+        ExecuteUserScriptError: If there is an exception raised when creating the process.
     """
 
     if capture_error:
         return_code, output, process = create(
             cmd,
-            error_class,
+            error_classes,
             processes_per_host,
             env=os.environ,
             cwd=cwd or environment.code_dir,
             capture_error=True,
             **kwargs,
         )
-        stderr = output[1]
+        stderr = " ".join(output)
```

Comment:
output is a list of size 2. The `output[0]` has stdout and `output[1]` has stderr.  Why are we combining both here? The `stdout` might contain normal log messages as well. Could you please clarify?

Reply:
output is the combination of watch function over stdout and stderr, and I think watch only add contents to output is an error is detected.
```
output = await asyncio.gather(
        watch(proc.stdout, error_classes, processes_per_host), watch(proc.stderr, error_classes, processes_per_host)
    )
```
so output should only contain error messages from both stderr and stdout, and it's mostly empty before this PR.

Normal processed logs are printed out here?
https://github.com/aws/sagemaker-training-toolkit/blob/master/src/sagemaker_training/process.py#L82